include/ofi_util.h

```diff
@@ -327,13 +327,28 @@ static inline void ofi_ep_rem_wr_cntr_inc(struct util_ep *ep)
 
 typedef void (*ofi_ep_cntr_inc_func)(struct util_ep *);
 
-static const ofi_ep_cntr_inc_func ofi_ep_cntr_inc_funcs[] = {
-	[FI_TRANSMIT] = ofi_ep_tx_cntr_inc,
-	[FI_RECV] = ofi_ep_rx_cntr_inc,
-	[FI_READ] = ofi_ep_rd_cntr_inc,
-	[FI_WRITE] = ofi_ep_wr_cntr_inc,
-	[FI_REMOTE_READ] = ofi_ep_rem_rd_cntr_inc,
-	[FI_REMOTE_WRITE] = ofi_ep_rem_wr_cntr_inc,
+static const ofi_ep_cntr_inc_func ofi_ep_tx_cntr_inc_funcs[] = {
+	[ofi_op_msg] = ofi_ep_tx_cntr_inc,
+	[ofi_op_tagged] = ofi_ep_tx_cntr_inc,
+	[ofi_op_read_req] = ofi_ep_rd_cntr_inc,
+	[ofi_op_read_rsp] = ofi_ep_rd_cntr_inc,
+	[ofi_op_write] = ofi_ep_wr_cntr_inc,
+	[ofi_op_write_rsp] = ofi_ep_wr_cntr_inc,
+	[ofi_op_atomic] = ofi_ep_wr_cntr_inc,
+	[ofi_op_atomic_fetch] = ofi_ep_rd_cntr_inc,
+	[ofi_op_atomic_compare] = ofi_ep_rd_cntr_inc,
+};
+
+static const ofi_ep_cntr_inc_func ofi_ep_rx_cntr_inc_funcs[] = {
+	[ofi_op_msg] = ofi_ep_rx_cntr_inc,
+	[ofi_op_tagged] = ofi_ep_rx_cntr_inc,
+	[ofi_op_read_req] = ofi_ep_rem_rd_cntr_inc,
+	[ofi_op_read_rsp] = ofi_ep_rem_rd_cntr_inc,
+	[ofi_op_write] = ofi_ep_rem_wr_cntr_inc,
+	[ofi_op_write_rsp] = ofi_ep_rem_wr_cntr_inc,
+	[ofi_op_atomic] = ofi_ep_rem_wr_cntr_inc,
+	[ofi_op_atomic_fetch] = ofi_ep_rem_rd_cntr_inc,
+	[ofi_op_atomic_compare] = ofi_ep_rem_rd_cntr_inc,
 };
```

Comment:
Please declare these arrays as extern and move into a .c file.  Otherwise, all files that include this header will get a copy of the arrays.

Reply:
Do you want me to do the same with the ofi_rx_flags and ofi_tx_flags while I'm at it?

Reply:
gurgle - yes, please.  I should have caught these earlier.